joomla/joomla-cms
# [4.3][GUIDEDTOURS] Cleanup Step model

Pull Request for Issue # .

### Summary of Changes
This PR makes some minor clean up to Step model from guided tours component :

1. Simplify code of `canEditState` method. We do not have to set tour_id for the record before checking the permission. The only case we have to do that is when adding a new step, but this is already being taken care by `getItem` method
2. Simplify `save` method. We do not have get tour_id from session and set it for step. The value for that field is already passed from hidden variable on step form (and we already validate and make sure it's available when a step is being added/editted)
3. Remove some one time used variable
4. Rename variable $tourID to $tourId to follow camel case naming convention

### Testing Instructions
1. Use Joomla 4.3 nightly build or a Joomla 4.3 installation which is having Tours Guided available
2. Apply patch. Try to add/edit step for a tour, make sure nothing is broken


### Actual result BEFORE applying this Pull Request
Works


### Expected result AFTER applying this Pull Request
Works with cleaner code


### Link to documentations
Please select:
- [] Documentation link for docs.joomla.org: <link>

- [] Pull Request link for manual.joomla.org: <link>
